ts/src/pro/toobit.ts

```diff
@@ -0,0 +1,563 @@
+        //                 p: "0.21804",
+        //                 q: "80",
+        //                 m: true,
+        //             },
+        //         ],
+        //         f: true,  // initial first snapshot or not
+        //         sendTime: 1757244002117,
+        //         shared: false,
+        //     }
+        //
+        const topic = this.safeString (message, 'topic');
+        // const isSnapshot = this.safeBool (message, 'f');
+        // if (this.handleErrorMessage (client, message)) {
+        //     return;
+        // }
+        // if (event === 'pong') {
+        //     client.lastPong = this.milliseconds ();
+        // }
+        const methods: Dict = {
+            'trade': this.handleTrades,
+            'kline': this.handleOHLCV,
+            'realtimes': this.handleTickers,
+            'depth': this.handlePartialOrderBook,
+        };
+        const method = this.safeValue (methods, topic);
+        if (method !== undefined) {
+            // if (isSnapshot) {
+            //     return; // todo
+            // }
+            method.call (this, client, message);
+        }
+    }
+
+    /**
+     * @method
+     * @name toobit#watchTrades
+     * @description watches information on multiple trades made in a market
+     * @see https://toobit-docs.github.io/apidocs/spot/v1/en/#trade-streams
+     * @param {string} symbol unified market symbol of the market trades were made in
+     * @param {int} [since] the earliest time in ms to fetch trades for
+     * @param {int} [limit] the maximum number of trade structures to retrieve
+     * @param {object} [params] extra parameters specific to the exchange API endpoint
+     * @returns {object[]} a list of [trade structures]{@link https://docs.ccxt.com/#/?id=trade-structure}
+     */
+    async watchTrades (symbol: string, since: Int = undefined, limit: Int = undefined, params = {}): Promise<Trade[]> {
+        return await this.watchTradesForSymbols ([ symbol ], since, limit, params);
+    }
+
+    /**
+     * @method
+     * @name toobit#watchTradesForSymbols
+     * @description get the list of most recent trades for a list of symbols
+     * @see https://toobit-docs.github.io/apidocs/spot/v1/en/#trade-streams
+     * @param {string[]} symbols unified symbol of the market to fetch trades for
+     * @param {int} [since] timestamp in ms of the earliest trade to fetch
+     * @param {int} [limit] the maximum amount of trades to fetch
+     * @param {object} [params] extra parameters specific to the exchange API endpoint
+     * @param {string} [params.name] the name of the method to call, 'trade' or 'aggTrade', default is 'trade'
+     * @returns {object[]} a list of [trade structures]{@link https://docs.ccxt.com/#/?id=public-trades}
+     */
+    async watchTradesForSymbols (symbols: string[], since: Int = undefined, limit: Int = undefined, params = {}): Promise<Trade[]> {
+        await this.loadMarkets ();
+        if (symbols === undefined) {
```

Comment:
you can use marketSymbols to do this check, also === undefined, does nto check .lenght === 0